homeassistant/components/viam/config_flow.py

```diff
@@ -0,0 +1,204 @@
+        return {"title": location.name, "hub": hub}
+
+    raise CannotConnect
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for viam."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize."""
+        self.credential_type = None
+        self.info = {}
+        self.data = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.credential_type = user_input[CONF_CREDENTIAL_TYPE]
+            return await self.async_step_auth()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_AUTH_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_auth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                self.info = await validate_input(
+                    self.hass, {"credential_type": self.credential_type, **user_input}
+                )
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                self.data = user_input
+                return await self.async_step_robot()
+
+        schema = STEP_AUTH_ROBOT_DATA_SCHEMA
+        if self.credential_type == "api-key":
+            schema = STEP_AUTH_ORG_DATA_SCHEMA
+
+        return self.async_show_form(
+            step_id="auth",
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_robot(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Select robot from location."""
+
+        locations = await self.info["hub"].client.app_client.list_locations()
+        robots = await self.info["hub"].client.app_client.list_robots(
+            next(iter(locations)).id
+        )
+        if user_input is not None:
+            robot_id = next(
+                robot.id for robot in robots if robot.name == user_input[CONF_ROBOT]
+            )
+            self.data.update(
+                {"robot_id": robot_id, "credential_type": self.credential_type}
+            )
+            self.info["hub"].close()
+            return self.async_create_entry(title=self.info["title"], data=self.data)
+
+        return self.async_show_form(
+            step_id="robot",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_ROBOT): selector(
+                        {"select": {"options": [robot.name for robot in robots]}}
```

Comment:
Please use the classes here too as you do it above. We should avoid using the bare strings